File: solutions/elixir/acronym/3/lib/acronym.ex
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    string
    |> String.split(~r/[\s-_]+/, trim: true)
    |> Enum.map(&String.first/1)
    |> List.to_string()
    |> String.upcase()
  end
end

File: solutions/ruby/acronym/3/acronym.rb
# frozen_string_literal: true

# Create acronyms from strings
module Acronym
  def self.abbreviate(str)
    str
      .split(/[\s\-_]+/)
      .map(&:chr)
      .join
      .upcase
  end

  def self.abbreviate_bis(str)
    str
      .scan(/\b\w/)
      .join
      .upcase
  end
end
